Misc fixes - byte array, minumum fee, etc (#353)
aeternity/protocol: consensus/consensus.md

### Coins and tokens

#### Aeternity tokens

We use the following denominations:

```

> 1                          | aetto

Reviewer: From https://en.wikipedia.org/wiki/Atto-
Ok.

  `1` (*10^-18) aeons.)
* Time to live (TTL).

*Note*: There is also a node configurable minimum gas price - this is the
minimum gas price a node accepts and is not under consensus. This value is

> normally higher than the minimal gas price as dictated by consensus. For

Reviewer 2: ```suggestion
higher or equal to the minimal gas price as dictated by consensus. For
```

| Channel offchain       |            0   |     0 |
| Contract create        | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(ContractCreateTx) * GasPerByte`. It also includes gas for contract execution. |
| Contract call          | `30 * BaseGas` | Proportional to the byte size of the transaction, specifically: `byte_size(ContractCallTx) * GasPerByte`. It also includes gas for contract execution. |
| GA Attach              | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAAttachTx) * GasPerByte`. It also includes gas for execution of the init function. |

> | GA Meta                | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAMetaTx) * GasPerByte`. It also includes gas for execution of the authentication function + gas for the wrapped transaction. |

Reviewer: Considering [`aetx:gas_limit`](https://github.com/aeternity/aeternity/blob/be71889c61c107cd848e97d3c58623ca0e93581b/apps/aetx/src/aetx.erl#L257-L258) and `aega_meta_tx:gas_limit`, I do not see how the base gas of the inner tx is not accounted for this gas (i.e. equivalent gas of tx for micro block gas limit).

Author: "gas for the wrapped transactions" is not very precise, it would and should include also base gas...

> | GA Meta                | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAMetaTx) * GasPerByte`. It also includes gas for execution of the authentication function + gas for the wrapped transaction. |

Author: I think that this is taken care of by https://github.com/aeternity/aeternity/pull/2362 when it comes to actual charging. But the `gas_limit` will indeed make use be a bit pessimistic when it comes to picking what candidate transactions may fit in a microblock. This could be made more precise but with some additional complexity.

> | GA Meta                | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAMetaTx) * GasPerByte`. It also includes gas for execution of the authentication function + gas for the wrapped transaction. |

Author: It should read `transactions` - it is obviously for transactions that have two signing parties. Also, while you may not add superfluous "signatures" (i.e. wrapping layers) it is actually allowed for the same account to sign (i.e wrap) multiple times though I see no real use case for it.

For channels there are several places where multiple wrapping may happen.

> | GA Meta                | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAMetaTx) * GasPerByte`. It also includes gas for execution of the authentication function + gas for the wrapped transaction. |

Reviewer: Tracked https://www.pivotaltracker.com/story/show/166018387

> | GA Meta                | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAMetaTx) * GasPerByte`. It also includes gas for execution of the authentication function + recursively gas corresponding to the wrapped transaction(s). |

Reviewer: ```suggestion
| GA Meta                | `5 * BaseGas`  | Proportional to the byte size of the transaction, specifically: `byte_size(GAMetaTx) * GasPerByte`. It also includes gas for execution of the authentication function + recursively gas corresponding to the wrapped transaction(s) (excluding the byte size portion - in order not to account for the size of wrapped transactions multiple times). |
```
